emloop/api.py

```diff
@@ -16,17 +16,24 @@
 from .main_loop import MainLoop
 
 
-def create_output_dir(config: dict, output_root: str, default_model_name: str='Unnamed') -> str:
+def create_output_dir(config: dict, output_root: str, default_model_name: str='Unnamed', output_dir: str='') -> str:
     """
     Create output_dir under the given ``output_root`` and
         - dump the given config to YAML file under this dir
         - register a file logger logging to a file under this dir
 
-    :param config: config to be dumped
+    :param config: config with model_name
     :param output_root: dir wherein output_dir shall be created
     :param default_model_name: name to be used when `model.name` is not found in the config
+    :param output_dir: if specified, create directory with output_dir path and return the path
     :return: path to the created output_dir
     """
+    if output_dir:
+        logging.info('\tOutput dir is: %s', output_dir)
+        os.makedirs(output_dir, exist_ok=True)
```

Comment:
```suggestion
        os.makedirs(path.join(output_root, output_dir), exist_ok=True)
```
?

Reply:
Well I expect the user to give the whole path to the `--output_dir` argument.. Not just the name of the `output_dir` within the `output_root`.

Reply:
Obviously, but that is a bit confusing behavior. The `--output_root` argument is required (but have a default). I do not expect it to be entirely ignored if and only if `--output_dir` is present.